src/chronify/time_series_mapper_index.py

```diff
@@ -0,0 +1,268 @@
+            self._to_time_config: DatetimeRange = self._to_schema.time_config  # type: ignore
+            self._datetime_mapper = MapperDatetimeToDatetime(
+                engine, metadata, self._to_schema, to_schema
+            )
+
+        self._engine = engine
+        self._metadata = metadata
+        self._time_based_data_adjustment = time_based_data_adjustment or TimeBasedDataAdjustment()
+        self._aligned = aligned
+
+        self._index_generator = IndexTimeRangeGenerator(self._from_time_config)
+
+        # adjusts for leapyear, but no dst
+        self._datetime_generator = DatetimeRangeGenerator(
+            self._to_time_config, self._time_based_data_adjustment
+        )
+
+        self._tz_naive: tuple[bool, bool] = (
+            self._from_time_config.is_time_zone_naive(),
+            self._to_time_config.is_time_zone_naive(),
+        )
+
+        # self._not_supported_mappings()
+
+    def check_schema_consistency(self) -> None:
+        # TODO: fail for interpolate fall_back_hour
+        pass
+
+    @property
+    def _align_local_clocktime(self) -> bool:
+        """Formerly known as Industrial time, flag indicating index time should map to a local clock time"""
+        return (
+            self._from_time_config.is_time_zone_naive()
+            and self._time_based_data_adjustment.daylight_saving_adjustment
+            == DaylightSavingAdjustment(spring_forward_hour="drop", fall_back_hour="duplicate")
+        )
+
+    def create_intermediate_schema(self) -> TableSchema:
+        time_kwargs = self._from_time_config.model_dump()
+        time_kwargs = dict(
+            filter(lambda k_v: k_v[0] in DatetimeRange.model_fields, time_kwargs.items())
+        )
+        time_kwargs["start"] = self._from_time_config.start_timestamp
+        time_kwargs["time_type"] = TimeType.DATETIME
+
+        time_config = DatetimeRange(**time_kwargs)
+
+        from_schema_kwargs = self._from_schema.model_dump()
+        from_schema_kwargs["name"] = "intermediate_mapping"
+        from_schema_kwargs["time_config"] = time_config
+
+        int_schema = TableSchema(**from_schema_kwargs)
+        return int_schema
+
+    def map_time(
+        self,
+        check_mapped_timestamps: bool = False,
+        **kwargs,
+    ) -> None:
+        """Convert time columns with from_schema to to_schema configuration."""
+        self.check_schema_consistency()
+
+        map_table_name = "map_table"
+        dfm = self._create_mapping()
+        mapping_schema = self._create_mapping_schema(map_table_name)
+
+        # if using datetime mapper, check timestamps in that mapper
+        tmp_check_mapped_timestamps = self._datetime_mapper is None
+
+        apply_mapping(
+            dfm,
+            mapping_schema,
+            self._from_schema,
+            self._to_schema,
+            self._engine,
+            self._metadata,
+            check_mapped_timestamps=tmp_check_mapped_timestamps,
```

Comment:
Suppose the user sets `aligned=True` and `check_mapped_timestamps=False`.  This will check the mapped timestamps. This is a bug, right?

I'll suggest refactoring this code so that there are independent code paths for aligned vs unaligned. It will be easier to reason about. There could be different `map_time` functions. Or, there could be two subclasses that inherit from the main index-mapper class.